src/main/sensors/gyroanalyse.c

```diff
@@ -261,53 +277,73 @@ static FAST_CODE_NOINLINE void gyroDataAnalyseUpdate(gyroAnalyseState_t *state,
-                    float cubedData = data * data * data;
-
-                    // add previous bin before first rise
-                    if (!fftIncreasing) {
-                        cubedData += prevData * prevData * prevData;
-
-                        fftIncreasing = true;
+                // only consider bins above an arbitrary size
+                if (data > dynamicFilterIgnore){
+                    if (fftIncreasing || data > prevData * dynamicFilterThreshold) {
+                        float cubedData = data * data * data;
+                        // add previous bin before first rise
+                        if (!fftIncreasing) {
+                            cubedData += prevData * prevData * prevData;
+                            fftIncreasing = true;
+                        }
+                        fftSum += cubedData;
+                        // calculate weighted index starting at 1, not 0
+                        fftWeightedSum += cubedData * (i + 1);
                     }
-
-                    fftSum += cubedData;
-                    // calculate weighted index starting at 1, not 0
-                    fftWeightedSum += cubedData * (i + 1);
                 }
             }
 
             // get weighted center of relevant frequency range (this way we have a better resolution than 31.25Hz)
             // if no peak, go to highest point to minimise delay
-            float centerFreq = dynNotchMaxCentreHz;
+            float centerFreq = DYN_NOTCH_MAX_CENTRE_HZ;
             float fftMeanIndex = 0;
-
             if (fftSum > 0) {
                 // idx was shifted by 1 to start at 1, not 0
                 fftMeanIndex = (fftWeightedSum / fftSum) - 1;
                 // the index points at the center frequency of each bin so index 0 is actually 16.125Hz
-                centerFreq = constrain(fftMeanIndex * fftResolution, DYN_NOTCH_MIN_CENTRE_HZ, dynNotchMaxCentreHz);
+                centerFreq = fftMeanIndex * FFT_RESOLUTION;
+            } else {
+            centerFreq = state->prevCenterFreq[state->updateAxis];
             }
-
+            state->prevCenterFreq[state->updateAxis] = centerFreq;
+            centerFreq = constrain(centerFreq, DYN_NOTCH_MIN_CENTRE_HZ, DYN_NOTCH_MAX_CENTRE_HZ);
             centerFreq = biquadFilterApply(&state->detectedFrequencyFilter[state->updateAxis], centerFreq);
-            centerFreq = constrain(centerFreq, DYN_NOTCH_MIN_CENTRE_HZ, dynNotchMaxCentreHz);
+            centerFreq = constrain(centerFreq, DYN_NOTCH_MIN_CENTRE_HZ, DYN_NOTCH_MAX_CENTRE_HZ);
             state->centerFreq[state->updateAxis] = centerFreq;
 
+
             if (state->updateAxis == 0) {
                DEBUG_SET(DEBUG_FFT, 3, lrintf(fftMeanIndex * 100));
             }
-            DEBUG_SET(DEBUG_FFT_FREQ, state->updateAxis, state->centerFreq[state->updateAxis]);
+            if (state->updateAxis == 0 || state->updateAxis == 1) {
```

Comment:
Can you use the defines for the axes please?

Reply:
I'd love to replace `(state->updateAxis == 0 || state->updateAxis == 1)` with the proper names but had no idea how... if you could show me I will fix it.